src/Database/PG/Query/Class.hs

```diff
@@ -523,118 +563,119 @@ newtype JSONB = JSONB Value
   deriving newtype (Hashable)
 
 instance ToPrepArg JSON where
-  toPrepVal (JSON j) = toPrepValHelper PTI.JSON PE.json_ast j
+  encoder (JSON j) = pure (PE.json_ast j)
+  oid = PTI.JSON
 
 instance ToPrepArg JSONB where
-  toPrepVal (JSONB j) = toPrepValHelper PTI.JSONB PE.jsonb_ast j
+  encoder (JSONB j) = pure (PE.jsonb_ast j)
+  oid = PTI.JSONB
 
 instance (ToPrepArg a) => ToPrepArg (Maybe a) where
-  toPrepVal (Just a) = toPrepVal a
-  -- FIX ME, the oid here should be particular to the type
-  toPrepVal Nothing = (PTI.Auto, Nothing)
+  encoder x = x >>= encoder
+  oid = oid @a
 
 instance (ToPrepArg a) => ToPrepArgs [a] where
-  toPrepArgs = map toPrepVal
+  toPrepArgs = map toPrepArg
 
 instance (ToPrepArg a) => ToPrepArgs (Identity a) where
   toPrepArgs (Identity a) =
-    [toPrepVal a]
+    [toPrepArg a]
```

Comment:
`ToPrepArgs` (plural) feels weird to me; it feels like, with the exception of tuples, it's just `foldMap toPrepArg` basically?